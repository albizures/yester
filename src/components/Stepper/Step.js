import { Dimensions, View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import React from 'react'

const Step = (props) => {
  const { children } = props
  const { width, height } = Dimensions.get('window')
  const portrait = height > width
  const style = {
    justifyContent: portrait ? 'flex-start' : 'flex-start',
    paddingTop: portrait ? 50 : 10,
    width,
    height,
  }

  return (
    <View style={[styles.container, style]}>
      {children}
    </View>
  )
}

Step.propTypes = {
  children: PropTypes.node.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
})

export default Step
