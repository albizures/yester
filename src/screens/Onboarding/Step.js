import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import { View, Image } from 'react-native'
import { Heading2, Heading4 } from '../../components'

export default class Step extends Component {
  static propTypes = {
    cover: Image.propTypes.source.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    children: PropTypes.node,
  }
  render () {
    const { cover, title, description, children } = this.props
    return (
      <Fragment>
        <View style={{alignItems: 'center', justifyContent: 'flex-start'}}>
          <Image source={cover} style={{marginBottom: 20}} />
          <Heading2 keyName={title} style={{marginHorizontal: 75, marginVertical: 20}} />
          <Heading4 keyName={description} style={{marginHorizontal: 75}} />
        </View>
        {children}
      </Fragment>
    )
  }
}
