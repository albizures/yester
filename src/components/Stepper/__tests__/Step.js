import React from 'react';
import renderer from 'react-test-renderer';

import Step from '../Step';
const Children = () => <div>test</div>;

let mockPortrait = true;

jest.mock('Dimensions', () => {
	return {
		get: () =>
			mockPortrait ? { width: 40, height: 50 } : { height: 40, width: 50 },
	};
});

describe('src/screens/Step.js', () => {
	it('should render a portrait version', () => {
		const tree = renderer
			.create(
				<Step>
					<Children />
				</Step>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe('when itÂ´s landscape mode', () => {
		beforeEach(() => {
			mockPortrait = false;
		});
		it('should render a landscape version', () => {
			const tree = renderer
				.create(
					<Step>
						<Children />
					</Step>,
				)
				.toJSON();
			expect(tree).toMatchSnapshot();
		});
	});
});
