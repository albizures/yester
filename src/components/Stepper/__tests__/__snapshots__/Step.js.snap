// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/Step.js should render a portrait version 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "height": 50,
        "justifyContent": "center",
        "paddingTop": 0,
        "width": 40,
      },
    ]
  }
>
  <div>
    test
  </div>
</View>
`;

exports[`src/screens/Step.js when itÂ´s landscape mode should render a landscape version 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "height": 40,
        "justifyContent": "flex-start",
        "paddingTop": 10,
        "width": 50,
      },
    ]
  }
>
  <div>
    test
  </div>
</View>
`;
